design/20190708.certificate-request-crd.md

```diff
@@ -206,6 +206,22 @@ ownership of the default pool of issuers in the cert-manager project.
 Until the mutating webhook is fully implemented, we will handle defaulting
 internally in the controller.
 
+### CertificateRequest Annotations
+In order for `CertificateRequest` controllers to resolve requests, extra
+information may be needed that is not present in the API Spec. To pass on this
+information, a set of one or more annotations should be defined, with reliable
+value pairs. These annotations should be considered optional. Any
+`CertificiateRequest` controller that rely on these to function should fallback
+gracefully or be marked as failed in the event a required annotation is missing.
```

Comment:
ref: my questions in the selfsigned PR, we currently mark as *pending* if the annotation is missing